ui/app/adapters/sync/association.js

```diff
@@ -82,9 +82,14 @@ export default class SyncAssociationAdapter extends ApplicationAdapter {
       const association = Object.values(resp.data.associated_secrets).find((association) => {
         return association.mount === data.mount && association.secret_name === data.secret_name;
       });
+
+      // generate an id if an association is found
+      // (an association may not be found if the secret is being unsynced)
+      const id = association ? serializer.generateId(association) : undefined;
```

Comment:
i thought about whether to make the fix here or to make the serializers `generateId` return `null` if an `undefined` association was passed in, but that felt wrong. i thought it'd be better to not pass in `undefined` values in the first place. 

open to feedback here!